docs: add what is angular topic (#40811)
angular/angular: aio/content/guide/what-is-angular.md

* A component-based framework for building scalable web applications
* A collection of well-integrated libraries that cover a wide variety of features, including routing, forms management, client-server communication, and more
* A suite of developer tools to help you develop, build, test, and update your code

When you build applications with Angular, you're taking advantage of a platform that can scale from single-developer projects to enterprise-level applications. Angular is designed to make updating as easy as possible, so you can take advantage of the latest developments with a minimum of effort. Best of all, the Angular ecosystem consists of a diverse group of over 1.7 million developers, library authors, and content creators.

<div class="alert is-helpful">

See the <live-example name="what-is-angular"></live-example> for a working example containing the code snippets in this guide.

</div>

{@a essentials}
## Angular applications: The essentials

This section explains the core ideas behind Angular. Understanding these ideas can help you design and build your applications more effectively.

{@a components}
### Components

> Components are the building blocks that compose an application. A component includes a TypeScript class, the `@Component()` decorator, an HTML template, and styles. The `@Component()` decorator specifies the following additional information:

Reviewer: ```suggestion
Components are the building blocks that compose an application. A component includes a TypeScript class with a `@Component()` decorator, an HTML template, and styles. The `@Component()` decorator specifies the following Angular-specific information:
```
* Technically, the component decorator is part of the class definition, so it would make sense to say "with" here instead of treating it as a separate entity
* The "additional" here was a bit awkward since template and styles are on both sides; I could see making this something like "Angular-specific" or just omitting the word entirely.

<code-example path="what-is-angular/src/app/logger.service.ts"></code-example>

Next, the `hello-world-di.component.ts` file defines an Angular component. This component contains a button that uses the `writeCount` function of the Logger class. To access that function, the `Logger` service is injected into the `HelloWorldDI` class by adding `private logger: Logger` to the constructor.

<code-example path="what-is-angular/src/app/hello-world-di/hello-world-di.component.ts"></code-example>

For more information about dependency injection and Angular, see the [Dependency injection in Angular](guide/dependency-injection) section.

{@a cli}

## Angular CLI

The Angular CLI is the fastest, easiest, and recommended way to develop Angular applications. The Angular CLI makes a number of tasks easy. Here are some examples:

<table>
<tr>
<td><a href="cli/build">ng build</a></td>
<td>Compiles an Angular app into an output directory.</td>
</tr>
<tr>
<td><a href="cli/serve">ng serve</a></td>

> <td>Builds and serves your application, rebuilding on file changes</td>

Reviewer 2: Period missing after second point.

> # What is Angular?

Reviewer: We should probably add a link to this doc from the introduction page

Reviewer: @aikidave I do think we should add a link at the least

This topic can help you understand Angular: what Angular is, what advantages it provides, and what you might expect as you start to build your applications.

Angular is a development platform, built on [TypeScript](https://www.typescriptlang.org/). As a platform, Angular includes:

* A component-based framework for building scalable web applications
* A collection of well-integrated libraries that cover a wide variety of features, including routing, forms management, client-server communication, and more

> * A suite of developer tools that to develop, build, test, and update your code

Reviewer 3: You forgot to add `help` after `that`.

When you build applications with Angular, you're taking advantage of a platform that can scale from single-developer projects to enterprise-level applications. Angular is designed to make updating as easy as possible, so you can take advantage of the latest developments with a minimum of effort. Best of all, the Angular ecosystem consists of a diverse group of over 1.7 million developers, library authors, and content creators.

<div class="alert is-helpful">

See the <live-example name="what-is-angular"></live-example> for a working example containing the code snippets in this guide.

</div>

{@a cli}

> ## Angular CLI

Reviewer: IMO I would put the CLI section after first-party libraries in favor of more quickly showing what Angular's application development model looks like, which I think will be what most people come to this page looking for. Putting the CLI section first would make sense for something task-based (since you'd need to use it), but I don't think it's necessary for this doc.

Notice the use of the square brackets--that syntax indicates that you're binding the property or attribute to a value in the component class.

You can also declare event listeners to listen for and respond to user actions such as keystrokes, mouse movements, clicks, and touches. You declare an event listener by specifying the event name in parenthesis:

<code-example path="what-is-angular/src/app/hello-world-bindings/hello-world-bindings.component.html" region="event-binding"></code-example>

The preceding example calls a method, which is defined in the component class:

<code-example path="what-is-angular/src/app/hello-world-bindings/hello-world-bindings.component.ts" region="method"></code-example>

The following is an example of interpolation and bindings within an Angular template:

<code-tabs linenums="true">
  <code-pane
    header="hello-world-bindings.component.ts"
    path="what-is-angular/src/app/hello-world-bindings/hello-world-bindings.component.ts">
  </code-pane>
  <code-pane
    header="hello-world-bindings.component.html"
    path="what-is-angular/src/app/hello-world-bindings/hello-world-bindings.component.html"
    linenums="false">
  </code-pane>
</code-tabs>

> You can add additional functionality to your templates through the use of [directives](guide/built-in-directives). The most popular directives in Angular are \*ngIf and \*ngFor. You can use directives to perform a variety of tasks, such as dynamically modifying the DOM structure. And you can also create your own custom directives to create great user experiences.

Reviewer: ```suggestion
You can add additional functionality to your templates through the use of [directives](guide/built-in-directives). The most popular directives in Angular are `*ngIf` and `*ngFor`. You can use directives to perform a variety of tasks, such as dynamically modifying the DOM structure. And you can also create your own custom directives to create great user experiences.
```
(I don't actually know if the escaping is needed)

> The following code is an example of the *ngIf directive.

Reviewer: ```suggestion
The following code is an example of the `*ngIf` directive.
```

<code-tabs linenums="true">
  <code-pane
    header="hello-world-ngif.component.ts"

>     path="what-is-angular/src/app/hello-world-ngif/hello-world-ngif.component.ts">

Reviewer: How strongly do you feel about every example being some form of "hello world" vs. using something more real-world representative?

Reviewer 4: I don't have any issues with an example being clearly marked as an example in situations like these. In tutorials, however, I think something more real-world is appropriate. Here, the focus is more on the ideas.

Reviewer: Ack. Personally I think that making every example some form of "HelloWorld" or "ExampleComponent" makes the examples harder to read/understand with the extra level of disconnect from doing something real. This is based purely on intuition though, so I could be completely wrong.

Reviewer 4: It's worth keeping an eye on. Might be interesting to ask in a survey!

  </code-pane>
  <code-pane
    header="hello-world-ngif.component.html"
    path="what-is-angular/src/app/hello-world-ngif/hello-world-ngif.component.html"
    linenums="false">
  </code-pane>
</code-tabs>

Angular's declarative templates allow you to cleanly separate your application's logic from its presentation. Templates are based on standard HTML, so they're easy to build, maintain, and update.

For more information on what you can do with templates, see the [Templates](guide/template-syntax) section.

> 

Reviewer 5: Just realized this, but what about a section on how Angular handles styles? Like HTML, CSS can give newer devs a frame of reference. For seasoned devs whose focus hasn't been on styles/design, the fact that the CSS for a particular component applies only to that component is a huge plus. CSS can be a chore for someone without a design+css background. And the fact that you can use pre-processors out of the box is super.

Reviewer 4: I'm willing to consider this for a future update, but I don't think we need to add it now. Let's file an issue and address it in a v2 update?

Reviewer: IMO this detail isn't central enough to the identify of Angular to belong in this high-level of a doc and makes more sense in a components-specific doc.

{@a di}
### Dependency injection

> Dependency injection allows you to declare the dependencies of your components and business logic without taking care of their instantiation. Instead, Angular handles the instantiation for you. This design pattern allows you to write more testable and flexible code. Even though understanding dependency injection is not critical to start using Angular, we strongly recommend it as a best practice and many aspects of Angular take advantage of it to some degree.

Reviewer: Personally I don't like the term "business logic" here since it doesn't have a very concrete meaning. Any super strong reason not to replace "your components and business logic" with "your TypeScript classes" ?

Reviewer 4: Business logic might resonate more with individuals who are not developers who might read this topic. Business decision makers, etc. But if we're assuming the primary audience are developers, your language may make more sense.

To illustrate how dependency injection works, consider the following example. The first file, `logger.service.ts`, defines a `Logger` class. This class contains a `writeCount` function that logs a number to the console.

<code-example path="what-is-angular/src/app/logger.service.ts"></code-example>

Next, the `hello-world-di.component.ts` file defines an Angular component. This component contains a button that uses the `writeCount` function of the Logger class. To access that function, the `Logger` service is injected into the `HelloWorldDI` class by adding `private logger: Logger` to the constructor.

<code-example path="what-is-angular/src/app/hello-world-di/hello-world-di.component.ts"></code-example>

> You'll find the Angular CLI a valuable tool for building out your applications.

Reviewer 3: Did you mean `Angular Dependency Injection` instead of `Angular CLI`

{@a 1p-libraries}
## First-party libraries

The section, [Angular applications: The essentials](#essentials), provides a brief overview of a couple of the key architectural elements you'll use when building Angular applications. But the many benefits of Angular really become apparent when your application grows and you want to add additional functions such as site navigation or user input. That's when you can leverage the Angular platform to incorporate one of the many first-party libraries that Angular provides.

Some of the libraries available to you include:
<table>
<tr>
<td><a href="guide/router">Angular Router</a></td>
<td>Advanced client-side navigation and routing based on Angular components. Supports lazy-loading, nested routes, custom path matching, and more.</td>
</tr>
<tr>
<td><a href="guide/forms-overview">Angular Forms</td>
<td>Uniform system for form participation and validation.</td>
<tr>
<td><a href="guide/http">Angular HttpClient</a></td>
<td>Robust HTTP client that can power more advanced client-server communication.</td>
</tr>
<tr>
<td><a href="guide/animations">Angular Animations</a></td>

> <td>Rich system for driving animations based on application state</td>

Reviewer 3: A `.` is needed at the end of the line
